### PR TITLE
70932 redirect to contact details

### PR DESCRIPTION
**Pull request by JohnPadgettCivica**

### CODE REVIEW

File: src/EA.Weee.Web/Areas/Producer/ViewModels/EditContactDetailsViewModel.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    using System.Collections.Generic;
+
+    public class EditContactDetailsViewModel
+    {
+        public ContactDetailsViewModel ContactDetails { get; set; } = new ContactDetailsViewModel();
+        public Guid DirectRegistrantId { get; set; }
+
+        public Guid OrganisationId { get; set; }
+
+        public List<AdditionalContactModel> AdditionalContactModels { get; set; }
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/Producer/ViewModels/EditContactDetailsViewModel.cs:15]: @stevelee-civica dont think you need this, this is for the organisation model.

File: src/EA.Weee.Web/Areas/Producer/Mappings/ToViewModel/EditOrganisationDetailsMap.cs [hunk 2]
```diff
@@ -69,5 +69,37 @@ private ExternalAddressData MapBusinessAddress(SmallProducerSubmissionData sourc
...
+            var externalOrganisationType = MapOrganisationType(source.OrganisationData.OrganisationType);
+            var businessAddressData = MapBusinessAddress(source);
+
+            var additionalContactDetails = source.CurrentSubmission.AdditionalCompanyDetailsData.Select(c => new AdditionalContactModel()
+            {
+                FirstName = c.FirstName,
+                LastName = c.LastName,
+            }).ToList();
+
+            var organisation = new OrganisationViewModel
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/Producer/Mappings/ToViewModel/EditOrganisationDetailsMap.cs:84]: @stevelee-civica  can remove any organisationViewModel related

File: src/EA.Weee.Web/Areas/Producer/Mappings/ToViewModel/EditOrganisationDetailsMap.cs [hunk 3]
```diff
@@ -69,5 +69,37 @@ private ExternalAddressData MapBusinessAddress(SmallProducerSubmissionData sourc
         {
             return mapper.Map<AddressData, ExternalAddressData>(source.CurrentSubmission.BusinessAddressData);
         }
+
+        public EditContactDetailsViewModel MapContactDetails(SmallProducerSubmissionData source)
+        {
+            var externalOrganisationType = MapOrganisationType(source.OrganisationData.OrganisationType);
+            var businessAddressData = MapBusinessAddress(source);
+
+            var additionalContactDetails = source.CurrentSubmission.AdditionalCompanyDetailsData.Select(c => new AdditionalContactModel()
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/Producer/Mappings/ToViewModel/EditOrganisationDetailsMap.cs:78]: @stevelee-civica dont need.

File: src/EA.Weee.Web/Areas/Producer/Mappings/ToViewModel/EditOrganisationDetailsMap.cs [hunk 4]
```diff
@@ -69,5 +69,37 @@ private ExternalAddressData MapBusinessAddress(SmallProducerSubmissionData sourc
...
+            };
+
+            var viewModel = new EditContactDetailsViewModel
+            {
+                DirectRegistrantId = source.DirectRegistrantId,
+                OrganisationId = source.OrganisationData.Id,
+                //Organisation = organisation,
+                AdditionalContactModels = additionalContactDetails,
+                HasAuthorisedRepresentitive = source.HasAuthorisedRepresentitive
+            };
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/Producer/Mappings/ToViewModel/EditOrganisationDetailsMap.cs:100]: @stevelee-civica  here you need to populate the ContactDetailsViewModel based on the SmallProducerSubmissionData.SmallProducerSubmissionHistoryData.ContactAddressData and ContactData properties.

File: src/EA.Weee.Web/Areas/Producer/Mappings/ToViewModel/EditOrganisationDetailsMap.cs [hunk 5]
```diff
@@ -69,5 +69,37 @@ private ExternalAddressData MapBusinessAddress(SmallProducerSubmissionData sourc
         {
             return mapper.Map<AddressData, ExternalAddressData>(source.CurrentSubmission.BusinessAddressData);
         }
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/Producer/Mappings/ToViewModel/EditOrganisationDetailsMap.cs:71]: @stevelee-civica  You shouldnt make any changes to this file.

File: src/EA.Weee.Web/Areas/Producer/Mappings/ToViewModel/EditContactDetailsMap.cs [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        {
+            this.mapper = mapper;
+        }
+
+        public EditContactDetailsViewModel Map(SmallProducerSubmissionData source)
+        {
+            var externalOrganisationType = MapOrganisationType(source.OrganisationData.OrganisationType);
+            var businessAddressData = MapBusinessAddress(source);
+
+            var additionalContactDetails = source.CurrentSubmission.AdditionalCompanyDetailsData.Select(c => new AdditionalContactModel()
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/Producer/Mappings/ToViewModel/EditContactDetailsMap.cs:26]: @stevelee-civica  
 here you need to populate the EditContactDetailsViewModel ContactDetailsViewModel based on the SmallProducerSubmissionData.SmallProducerSubmissionHistoryData.ContactAddressData and ContactData properties.

File: src/EA.Weee.Web/Areas/Producer/Mappings/ToViewModel/EditOrganisationDetailsMap.cs (hunk 5 above)
Comment (stevelee-civica) [reply src/EA.Weee.Web/Areas/Producer/Mappings/ToViewModel/EditOrganisationDetailsMap.cs]: restored

File: src/EA.Weee.Web/Areas/Producer/ViewModels/EditContactDetailsViewModel.cs (hunk 1 above)
Comment (stevelee-civica) [reply src/EA.Weee.Web/Areas/Producer/ViewModels/EditContactDetailsViewModel.cs:15]: removed

File: src/EA.Weee.Web/Areas/Producer/Views/ProducerSubmission/EditContactDetails.cshtml [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+@using EA.Weee.Web.ViewModels.Shared
+@using EnumHelper = EA.Prsd.Core.Helpers.EnumHelper
+@model EA.Weee.Web.Areas.Producer.ViewModels.EditContactDetailsViewModel
+@{
+    const string title = "Contact details";
+    ViewBag.Title = title + " - GOV.UK";
+}
+<style>
+
+    .flex-container {
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/Producer/Views/ProducerSubmission/EditContactDetails.cshtml:23]: @stevelee-civica  Dont think you need these styles in this view.

File: src/EA.Weee.Web/Areas/Producer/Views/ProducerSubmission/EditContactDetails.cshtml [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+            @Html.Gds().ValidationMessageFor(m => m.ContactDetails.AddressData.Telephone)
+            @Html.Gds().TextBoxFor(m => m.ContactDetails.AddressData.Telephone, new { @maxlength = CommonMaxFieldLengths.Telephone })
+        </div>
+
+        // Contact address
+
+        <h2 class="govuk-heading-s">Contact address</h2>
+
+        <div class="govuk-body">
+            For non-UK addresses, enter your details manually.
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/Producer/Views/ProducerSubmission/EditContactDetails.cshtml:96]: @stevelee-civica  can remove this as there is no lookup.

File: src/EA.Weee.RequestHandlers.Tests.Unit/Organisations/DirectRegistrants/EditContactDetailsRequestHandlerTests.cs [hunk 9]
```diff
@@ -81,20 +83,6 @@ await Assert.ThrowsAsync<SecurityException>(
                 async () => await handler.HandleAsync(request));
         }
 
-        [Fact]
-        public async Task HandleAsync_EnsureOrganisationAccess_IsCalled()
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.RequestHandlers.Tests.Unit/Organisations/DirectRegistrants/EditContactDetailsRequestHandlerTests.cs]: @stevelee-civica  this test should still be valid.

**Pull request merged**